Bottom: b7b33874964645a205c971809ceb2b241a6dc42e
Top:    ff08162da6f91a8872d8b5c41f2d638aa908357f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:22:52 +0000

Remove graphics pointer from HID structure


---

diff --git a/src/hid.h b/src/hid.h
index ed921db..7943949 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -294,8 +294,6 @@ typedef enum
     void (*notify_crosshair_change) (bool changes_complete);
     void (*notify_mark_change) (bool changes_complete);
 
-    HID_DRAW *graphics;
-
     /* This is for the printer.  If you call this for the GUI, xval and
        yval are ignored, and a dialog pops up to lead you through the
        calibration procedure.  For the printer, if xval and yval are
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index 2e7bdc2..72c49e0 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -359,8 +359,6 @@ hid_batch_init ()
   batch_hid.attribute_dialog      = batch_attribute_dialog;
   batch_hid.show_item             = batch_show_item;
 
-  batch_hid.graphics              = &batch_graphics;
-
   common_nogui_graphics_class_init (&batch_graphics_class);
   common_draw_helpers_class_init (&batch_graphics_class);
 
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 7014af5..86abb15 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -182,8 +182,6 @@ hid_extents_init (void)
   extents_hid.name                = "extents-extents";
   extents_hid.description         = "used to calculate extents";
 
-  extents_hid.graphics            = &extents_graphics;
-
   common_draw_helpers_class_init (&extents_graphics_class);
 
   extents_graphics_class.set_layer      = extents_set_layer;
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 2618d98..a746411 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -523,7 +523,6 @@ hid_nogui_get_hid (void)
   nogui_hid.name        = "nogui";
   nogui_hid.description = "Default GUI when no other GUI is present.  "
                           "Does nothing.";
-  nogui_hid.graphics    = &nogui_graphics;
 
   common_nogui_init (&nogui_hid);
 
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index f162783..eb5a2d5 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1626,8 +1626,6 @@ hid_gcode_init ()
   gcode_hid.calibrate           = gcode_calibrate;
   gcode_hid.set_crosshair       = gcode_set_crosshair;
 
-  gcode_hid.graphics            = &gcode_graphics;
-
   common_nogui_graphics_class_init (&gcode_graphics_class);
   common_draw_helpers_class_init (&gcode_graphics_class);
 
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 0f8f287..60627b7 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1358,8 +1358,6 @@ hid_gerber_init ()
   gerber_hid.calibrate           = gerber_calibrate;
   gerber_hid.set_crosshair       = gerber_set_crosshair;
 
-  gerber_hid.graphics            = &gerber_graphics;
-
   common_nogui_graphics_class_init (&gerber_graphics_class);
   common_draw_helpers_class_init (&gerber_graphics_class);
 
diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 705b45b..90fbd4b 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1251,7 +1251,7 @@ ghid_request_debug_draw (void)
 {
   /* No special setup requirements, drawing goes into
    * the backing pixmap. */
-  return ghid_hid.graphics;
+  return &ghid_graphics;
 }
 
 void
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 32a125d..70ec294 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1365,7 +1365,7 @@ ghid_request_debug_draw (void)
                 port->view.flip_y ? port->view.y0 - PCB->MaxHeight :
                              -port->view.y0, 0);
 
-  return ghid_hid.graphics;
+  return &ghid_graphics;
 }
 
 void
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 44202cd..955d6f4 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2183,8 +2183,6 @@ hid_gtk_init ()
   ghid_hid.notify_save_pcb          = ghid_notify_save_pcb;
   ghid_hid.notify_filename_changed  = ghid_notify_filename_changed;
 
-  ghid_hid.graphics                 = &ghid_graphics;
-
   common_nogui_graphics_class_init (&ghid_graphics_class);
   common_draw_helpers_class_init (&ghid_graphics_class);
 
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 106465f..8cd3a08 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4032,7 +4032,7 @@ lesstif_request_debug_draw (void)
 {
   /* Send drawing to the backing pixmap */
   pixmap = main_pixmap;
-  return lesstif_hid.graphics;
+  return &lesstif_graphics;
 }
 
 static void
@@ -4113,8 +4113,6 @@ hid_lesstif_init ()
   lesstif_hid.flush_debug_draw        = lesstif_flush_debug_draw;
   lesstif_hid.finish_debug_draw       = lesstif_finish_debug_draw;
 
-  lesstif_hid.graphics                = &lesstif_graphics;
-
   common_nogui_graphics_class_init (&lesstif_graphics_class);
   common_draw_helpers_class_init (&lesstif_graphics_class);
 
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index a460e95..da3179b 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -1077,8 +1077,6 @@ hid_nelma_init()
   nelma_hid.calibrate           = nelma_calibrate;
   nelma_hid.set_crosshair       = nelma_set_crosshair;
 
-  nelma_hid.graphics            = &nelma_graphics;
-
   common_nogui_graphics_class_init (&nelma_graphics_class);
   common_draw_helpers_class_init (&nelma_graphics_class);
 
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 8a72345..3b16696 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1852,8 +1852,6 @@ hid_png_init ()
   png_hid.calibrate           = png_calibrate;
   png_hid.set_crosshair       = png_set_crosshair;
 
-  png_hid.graphics            = &png_graphics;
-
   common_nogui_graphics_class_init (&png_graphics_class);
   common_draw_helpers_class_init (&png_graphics_class);
 
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 3f46376..0d54ad6 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -693,8 +693,6 @@ hid_eps_init ()
   eps_hid.calibrate           = eps_calibrate;
   eps_hid.set_crosshair       = eps_set_crosshair;
 
-  eps_hid.graphics            = &eps_graphics;
-
   common_nogui_graphics_class_init (&eps_graphics_class);
   common_draw_helpers_class_init (&eps_graphics_class);
 
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index 248831f..6c8842c 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1571,8 +1571,6 @@ hid_ps_init ()
   ps_hid.description        = N_("Postscript export");
   ps_hid.exporter           = 1;
 
-  ps_hid.graphics           = &ps_graphics;
-
   common_nogui_graphics_class_init (&ps_graphics_class);
   common_draw_helpers_class_init (&ps_graphics_class);
   ps_ps_graphics_class_init (&ps_graphics_class);
